app/utilities/schema.py

```diff
@@ -1,37 +1,49 @@
-from app import cache
-from app.parser.v0_0_1.schema_parser import SchemaParser
-from app.schema_loader.schema_loader import load_schema
+import json
+import os
+
+from structlog import get_logger
 
+from app import cache
+from app import settings
 
-def get_schema(metadata):
-    """
-    Get the schema for the current user
-    :return: (json, schema) # Tuple of json and schema object from schema file
-    """
 
-    eq_id = metadata["eq_id"]
-    form_type = metadata["form_type"]
-    language_code = metadata["language_code"] if "language_code" in metadata else None
+logger = get_logger()
 
-    json_schema = load_and_parse_schema(eq_id, form_type, language_code)
 
-    return json_schema
+@cache.memoize()
+def load_schema_from_metadata(metadata):
+    return load_schema_from_params(metadata['eq_id'], metadata['form_type'], metadata.get('language_code'))
 
 
 @cache.memoize()
-def load_and_parse_schema(eq_id, form_type, language_code):
-    """
-    Use the schema loader to get the schema from disk. Then use the parse to construct the object schema
-    :param eq_id: the id of the questionnaire
-    :param form_type: the form type
-    :return: (json, schema) # Tuple of json and schema object from schema file
-    """
-    # load the schema
-
-    json_schema = load_schema(eq_id, form_type, language_code)
-    if json_schema:
-        parser = SchemaParser(json_schema)
-        parser.parse()
-        return json_schema
-    else:
-        raise ValueError("No schema available")
+def load_schema_from_params(eq_id, form_type, language_code='en'):
+    return load_schema_file("{}_{}.json".format(eq_id, form_type), language_code)
+
+
+def load_schema_file(schema_file, language_code='en'):
+    logger.info("loading schema", filename=schema_file, language_code=language_code)
+
+    try:
+        if language_code is not None and language_code != 'en':
```

Comment:
Fair enough. didn't realise that None would override the default. However if we are defaulting in this function maybe we should not explicitly pass in None if the language isn't in their metadata?